apps/ejabberd/src/mongoose_commands.erl

```diff
@@ -117,6 +118,7 @@
           args = [] :: [argspec()],                   %% this is both for introspection and type check on call
           caller_pos :: integer(),                    %% internal use
           identifiers = [] :: [atom()],               %% resource identifiers, a subset of args
+          queryparams = [] :: [atom()],               %% query parameters, a subset of args
```

Comment:
In any order - yes, they are sorted later. Optional - no, our command engine has no notion of optional arguments. Perhaps it should, it wouldn't be difficult to implement.